crates/rpc/rpc-eth-api/src/helpers/call.rs

```diff
@@ -776,4 +779,428 @@ pub trait Call: LoadState + SpawnBlocking {
+                value,
+                input,
+                nonce,
+                access_list,
+                chain_id,
+                blob_versioned_hashes,
+                max_fee_per_blob_gas,
+                ..
+            } = request;
+
+            let CallFees { max_priority_fee_per_gas, gas_price, max_fee_per_blob_gas } =
+                CallFees::ensure_fees(
+                    gas_price.map(U256::from),
+                    max_fee_per_gas.map(U256::from),
+                    max_priority_fee_per_gas.map(U256::from),
+                    block_env.basefee,
+                    blob_versioned_hashes.as_deref(),
+                    max_fee_per_blob_gas.map(U256::from),
+                    block_env.get_blob_gasprice().map(U256::from),
+                )?;
+
+            let gas_limit =
+                gas.unwrap_or_else(|| block_env.gas_limit.min(U256::from(u64::MAX)).to());
+            let env = TxEnv {
+                gas_limit: gas_limit
+                    .try_into()
+                    .map_err(|_| RpcInvalidTransactionError::GasUintOverflow)?,
+                nonce,
+                caller: from.unwrap_or_default(),
+                gas_price,
+                gas_priority_fee: max_priority_fee_per_gas,
+                transact_to: to.unwrap_or(TxKind::Create),
+                value: value.unwrap_or_default(),
+                data: input.try_into_unique_input()?.unwrap_or_default(),
+                chain_id,
+                access_list: access_list
+                    .map(reth_rpc_types::AccessList::into_flattened)
+                    .unwrap_or_default(),
+                // EIP-4844 fields
+                blob_hashes: blob_versioned_hashes.unwrap_or_default(),
+                max_fee_per_blob_gas,
+                #[cfg(feature = "optimism")]
+                optimism: OptimismFields { enveloped_tx: Some(Bytes::new()), ..Default::default() },
+            };
+
+            Ok(env)
+        }
+        let tx = create_txn_env(&block, request)?;
+        Ok(EnvWithHandlerCfg::new_with_cfg_env(cfg, block, tx))
+    }
+
+    /// Prepares the [`EnvWithHandlerCfg`] for execution.
+    ///
+    /// Does not commit any changes to the underlying database.
+    ///
+    /// EVM settings:
+    ///  - `disable_block_gas_limit` is set to `true`
+    ///  - `disable_eip3607` is set to `true`
+    ///  - `disable_base_fee` is set to `true`
+    ///  - `nonce` is set to `None`
+    fn prepare_call_env<DB>(
+        &self,
+        mut cfg: CfgEnvWithHandlerCfg,
+        mut block: BlockEnv,
+        request: TransactionRequest,
+        gas_limit: u64,
+        db: &mut CacheDB<DB>,
+        overrides: EvmOverrides,
+    ) -> EthResult<EnvWithHandlerCfg>
+    where
+        DB: DatabaseRef,
+        EthApiError: From<<DB as DatabaseRef>::Error>,
+    {
+        /// Applies the given block overrides to the env
+        fn apply_block_overrides(overrides: BlockOverrides, env: &mut BlockEnv) {
```

Comment:
this function we can keep as a standalone function

Reply:
Fixed. Thanks.